lib/memstore.ts

```diff
@@ -48,27 +17,66 @@ export type MemoryCookieStoreIndex = {
   }
 }
 
+/**
+ * An in-memory {@link Store} implementation for {@link CookieJar}. This is the default implementation used by
+ * {@link CookieJar} and supports both async and sync operations. Also supports serialization, getAllCookies, and removeAllCookies.
```

Comment:
Should we warn "this is mostly a development tool, you should use something better in production"?

Reply:
Does that even need to be stated? Also, there are some cases where you might have a use for an in-memory store in production. For example, if you are an automation tool that needs to login to a site and store an auth cookie.